modules/viz/include/opencv2/viz/types.hpp

```diff
@@ -119,6 +119,12 @@ namespace cv
         class CV_EXPORTS Mesh
         {
         public:
+            enum FileTypeReader
+            {
+                PLY,
+                OBJ
+            };
```

Comment:
Btw, I like the method with enum (auto, ply, obj) much more, please don't hurry to revert this.